.github/workflows/main.yml

```diff
@@ -15,30 +15,41 @@ on:
     branches:
       - '**'
 
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
+
 jobs:
   tests:
     name: ${{ matrix.os }} / ${{ matrix.python-version }} ${{ matrix.suffix }}
-    runs-on: ${{ matrix.os }}-latest
-    continue-on-error: ${{ matrix.experimental }}
+    runs-on: ${{ matrix.image }}
     strategy:
       matrix:
-        os: [Ubuntu, MacOS, Windows]
+        os: [Ubuntu, macOS, Windows]
         python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
-        experimental: [false]
         bootstrap-args: [""]
         suffix: [""]
         include:
           - os: Ubuntu
+            image: ubuntu-latest
```

Comment:
Figured out a way to elegantly include updated OS images (not just blanket `-latest` images).